invenio_rdm_records/services/components/metadata.py

```diff
@@ -19,6 +20,9 @@ class MetadataComponent(ServiceComponent):
 
     field = "metadata"
     new_version_skip_fields = ["publication_date", "version"]
+    new_version_generated_fields = {
+        "publication_date": lambda draft, record: datetime.today().isoformat(),
```

Comment:
To avoid time zone discrepancies?
```suggestion
        "publication_date": lambda draft, record: datetime.today(timezone.utc).isoformat(),
```